src/Api/HL/Controller/CoreController.php

```diff
@@ -0,0 +1,576 @@
+        foreach ($methods as $method) {
+            $route_matches = $router->match($request->withMethod($method));
+            // Filter out this route
+            $route_matches = array_filter($route_matches, static function ($route) {
+                $controller = $route->getController();
+                $controller_method = $route->getMethod();
+                return !($controller === __CLASS__ && $controller_method->getShortName() === 'defaultRoute');
+            });
+            if (count($route_matches) > 0) {
+                $allowed_methods[] = $method;
+            }
+        }
+        sort($allowed_methods);
+        return $allowed_methods;
+    }
+
+    #[Route('/{req}', ['GET', 'POST', 'PATCH', 'PUT', 'DELETE'], ['req' => '.*'], -1)]
+    #[Doc\Route(
+        description: 'A fallback for when no other endpoint matches the request. A 404 error will be shown.',
+        methods: ['GET', 'POST', 'PATCH', 'PUT', "DELETE"],
+        responses: [
+            '200' => [
+                'description' => 'Never returned',
+            ],
+            '404' => [
+                'description' => 'No route found for the requested path',
+                'schema' => [
+                    'type' => Doc\Schema::TYPE_OBJECT,
+                    'properties' => []
+                ]
+            ]
+        ]
+    )]
+    public function defaultRoute(Request $request): Response
+    {
+        return new JSONResponse(null, 404);
+    }
+
+    #[Route(path: '/{req}', methods: ['OPTIONS'], requirements: ['req' => '.*'], priority: -1, security_level: Route::SECURITY_NONE)]
+    #[Doc\Route(
+        description: 'A global route that enables the OPTIONS method on all endpoints. This responds with an Accept header indicating which methods are allowed.',
+        methods: ['OPTIONS']
+    )]
+    public function defaultOptionsRoute(Request $request): Response
+    {
+        $authenticated = Session::getLoginUserID() !== false;
+        $allowed_methods = $authenticated ? $this->getAllowedMethodsForMatchedRoute($request) : ['GET', 'POST', 'PATCH', 'PUT', "DELETE"];
+        if (count($allowed_methods) === 0) {
+            return new JSONResponse(null, 404);
+        }
+        $response_headers = [];
+        if ($authenticated) {
+            $response_headers['Allow'] = $allowed_methods;
+        }
+        if (isset($_SERVER['HTTP_ACCESS_CONTROL_REQUEST_METHOD'])) {
+            $response_headers['Access-Control-Allow-Methods'] = $allowed_methods;
+        }
+        if (isset($_SERVER['HTTP_ACCESS_CONTROL_REQUEST_HEADERS'])) {
+            $response_headers['Access-Control-Allow-Headers'] = [
+                'Content-Type', 'Authorization', 'Origin', 'Accept', 'Glpi-Session-Token', 'Glpi-User-Token'
+            ];
+        }
+        return new JSONResponse(null, 204, $response_headers);
+    }
+
+    #[Route(path: '/session', methods: ['POST'], security_level: Route::SECURITY_NONE, tags: ['Session'])]
```

Comment:
Almost all routes have fist char in uppercase, except in this controller. They should probably use same naming rules.
```suggestion
    #[Route(path: '/Session', methods: ['POST'], security_level: Route::SECURITY_NONE, tags: ['Session'])]
```

Reply:
Route paths are case insensitive, but for consistency, I've fixed this.